File: 프로그래머스/0/120847. 최댓값 만들기 （1）/최댓값 만들기 （1）.py
def solution(numbers):
    answer = 0  # 최댓값을 저장할 변수

    numbers.sort()
    answer = numbers[-1] * numbers[-2]  # 오름차순 정렬 후 최댓값 두 개의 곱

    return answer  # 최댓값 반환

# 테스트 예제 실행
print(solution([1, 2, 3, 4, 5]))        # 20
print(solution([0, 31, 24, 10, 1, 9]))  # 744

File: 프로그래머스/0/120847. 최댓값 만들기 （1）/최댓값 만들기 （1）.go
// 배열에서 가장 큰 두 수의 곱을 반환하는 Go 함수
func solution(numbers []int) int {
	n := len(numbers)
	// 선택 정렬 (오름차순)
	for i := 0; i < n-1; i++ {
		for j := i + 1; j < n; j++ {
			if numbers[j] < numbers[i] {
				numbers[i], numbers[j] = numbers[j], numbers[i]
			}
		}
	}
	// 마지막 두 수의 곱이 최댓값
	return numbers[n-1] * numbers[n-2]
}
